src/core/server/deprecations/deprecations_service.ts

```diff
@@ -139,7 +150,13 @@ export class DeprecationsService implements CoreService<InternalDeprecationsServ
     };
   }
 
-  public start() {}
+  public start(): DeprecationsServiceStart {
+    return {
+      getAllDeprecations: (dependencies: GetDeprecationsContext) =>
+        this.deprecationsFactory.getAllDeprecations(dependencies),
+    };
```

Comment:
Instead of requiring the calling plugin to provide these dependencies, I think we should add this to our `RequestHandlerContext` and automatically wire up the ES client and SO client for the current request. This would be more in line with how we expose other services.

The type for `RequestHandlerContext` is defined in `src/core/server/index.ts` and implementation for our context provider is available in `src/core/server/core_route_handler_context.ts`. I'd expect that we add a new `deprecations` key under `RequestHandlerContext['core']` and use the lazy-instantiation approach we're using for the other clients in the implementation.

If we go with this approach, I don't believe we'll need to expose this 'raw' API to plugins and can keep this API internal (don't include it in `plugin_context.ts`).

Reply:
👍 Thanks @joshdover for the feedback! Can you let me know if this is what you had in mind: https://github.com/elastic/kibana/pull/105998/commits/993773e1db996b81eecdb58b64d64fc22a7c5576? I will follow up with updating tests and docs in a separate commit if you're happy with the approach.

Reply:
Yep, this looks perfect! Feel free to ping me again when the tests are in place. Thanks for this!